tools/extra/pac/fpgabist/bist_common.py

```diff
@@ -54,21 +54,26 @@ def check_required_cmds():
         sys.exit("Failed to find required BIST commands\nTerminating BIST")
 
 
-def get_afu_id(gbs_path=""):
+def get_afu_id(gbs_path="", bdf=None):
     if os.path.isfile(gbs_path):
         cmd = ["packager", "gbs-info", "--gbs={}".format(gbs_path)]
         output = subprocess.Popen(cmd, stdout=subprocess.PIPE)
         json_data = json.loads(output.communicate()[0])
         accel_data = json_data["afu-image"]["accelerator-clusters"]
         uuid = accel_data[0]["accelerator-type-uuid"].encode("ascii")
         return uuid.lower().replace("-", "")
-    else:
-        bdf_pattern = re.compile(BDF_PATTERN)
-        for fpga in glob.glob('/sys/class/fpga/*'):
+    elif bdf:
+        pattern = '{:02x}:{:02x}.{:x}'.format(bdf['bus'], bdf['device'],
```

Comment:
Would it be easier to rename the macro BDF_PATTERN with this pattern you are looking for here?
Did you find an issue using the other pattern or is this just a cleaner regex?

Reply:
Ok. I'll do that in the next round with the DMA update.